cmd/neofs-cli/internal/client/client.go

```diff
@@ -336,28 +403,54 @@ type DeleteObjectPrm struct {
 
 // DeleteObjectRes groups resulting values of DeleteObject operation.
 type DeleteObjectRes struct {
-	cliRes *client.ObjectDeleteRes
+	addrTombstone *addressSDK.Address
 }
 
 // TombstoneAddress returns address of the created object with tombstone.
 func (x DeleteObjectRes) TombstoneAddress() *addressSDK.Address {
-	return x.cliRes.TombstoneAddress()
+	return x.addrTombstone
 }
 
 // DeleteObject marks object to be removed from NeoFS through tombstone placement.
 //
 // Returns any error prevented the operation from completing correctly in error return.
-func DeleteObject(prm DeleteObjectPrm) (res DeleteObjectRes, err error) {
-	var delPrm client.DeleteObjectParams
+func DeleteObject(prm DeleteObjectPrm) (*DeleteObjectRes, error) {
+	var delPrm client.PrmObjectDelete
 
-	delPrm.WithAddress(prm.objAddr)
+	if id := prm.objAddr.ContainerID(); id != nil {
+		delPrm.FromContainer(*id)
+	}
 
-	res.cliRes, err = prm.cli.DeleteObject(context.Background(), &delPrm, append(prm.opts,
-		client.WithSession(prm.sessionToken),
-		client.WithBearer(prm.bearerToken),
-	)...)
+	if id := prm.objAddr.ObjectID(); id != nil {
+		delPrm.ByID(*id)
+	}
 
-	return
+	if prm.sessionToken != nil {
+		delPrm.WithinSession(*prm.sessionToken)
+	}
+
+	if prm.bearerToken != nil {
+		delPrm.WithBearerToken(*prm.bearerToken)
+	}
+
+	cliRes, err := prm.cli.ObjectDelete(context.Background(), delPrm)
+	if err != nil {
+		return nil, fmt.Errorf("remove object via client: %w", err)
```

Comment:
i remember that you dont like "error" words but I want to mention that "failed" or "could not" looks more clear for understanding in the logs in my opinion

Reply:
Calling context can do it, here it is a fail by design of `error` type.